src/adservice/build.gradle

```
@@ -60,14 +84,20 @@ protobuf {
             artifact = "io.grpc:protoc-gen-grpc-java:${grpcVersion}"
         }
     }
-    generateProtoTasks {
+    generateProtoTasks { task ->
         all()*.plugins {
             grpc {}
         }
         ofSourceSet('main')
     }
 }
 
+afterEvaluate {
+    // Ensure protos are copy before classes are generated
+    tasks.getByName('processResources').dependsOn('copyProtos')
+    tasks.getByName('generateProto').dependsOn 'copyProtos'
```

just to match line above
```suggestion
    tasks.getByName('generateProto').dependsOn('copyProtos')
```